test/integration_test/utils/k8s.go

```diff
@@ -124,6 +145,31 @@ func CreateObjects(objects []runtime.Object) error {
 	return nil
 }
 
+// ValidateApplicationWorkload checks application deployments and PVCs are healthy
+func ValidateApplicationWorkload(objects []runtime.Object, timeout, interval time.Duration) error {
+	for _, obj := range objects {
+		var err error
+		if dp, ok := obj.(*appsv1.Deployment); ok {
+			err = appops.Instance().ValidateDeployment(dp, timeout, interval)
+		} else if pvc, ok := obj.(*v1.PersistentVolumeClaim); ok {
+			t := func() (interface{}, bool, error) {
+				livePVC, err := coreops.Instance().GetPersistentVolumeClaim(pvc.Name, pvc.Namespace)
+				if err != nil {
+					return nil, true, err
+				} else if livePVC.Status.Phase != v1.ClaimBound {
+					return nil, true, fmt.Errorf("failed to validate PVC %s/%s, phase %s", pvc.Namespace, pvc.Name, pvc.Status.Phase)
+				}
+				return nil, false, nil
+			}
+			_, err = task.DoRetryWithTimeout(t, timeout, interval)
+		}
+		if err != nil {
+			return err
+		}
```

Comment:
We can use ValidatePersistentVolumeClaim from sched-ops here - https://github.com/portworx/sched-ops/blob/master/k8s/core/persistentvolumeclaims.go#L94.

Reply:
thanks for the hint. I was surprised I didn't find any PVC validation utilities.